### PR TITLE
Add ability to specify custom names in the OAuth login buttons

### LINKED ISSUE #154: [Discussion] Gitea/Forgejo
I don't really know the best way to handle this...but thought I would bring it up for discussion.

---

## Context
[Forgejo](https://codeberg.org/forgejo/forgejo) is a soft-fork of [Gitea](https://github.com/go-gitea/gitea). They provide [a justification](https://forgejo.org/faq/#why-was-forgejo-created) for existing:

> **Why was Forgejo created?**
> 
> In October 2022 the domains and trademark of Gitea were transferred to a for-profit company without knowledge or approval of the community. Despite [writing an open letter](https://gitea-open-letter.coding.social/), the takeover was later confirmed. Forgejo was created as an alternative providing a software forge whose governance further the interest of the general public.
> 
> Forgejo was initially presented as a “soft-fork” of Gitea, similar to [LineageOS](https://lineageos.org/), a community led distribution based on Android from Google. It is however better described as a product built on top of Gitea, Git and hundreds of other Free Software projects. Forgejo also has its own documentation, infrastructure, release pipeline, CI infrastructure, distribution channels etc.

## Why did you open this discussion issue, @wkeiuluf?
As you may or may not know, users aren't always the brightest. Seeing "login with gitea" might be confusing to a user (not the admin setting everything up, but one of their users) that is like "hey, what's gitea?"

Basically, I think it would be nice to have an aliased set of `GITEA` stuff but with `FORGEJO` in place. But that might not be the cleanest way to handle it. Hell, people reading this may even just think "you're dumb, who cares" and that's fine, too; that's why I opened up a discussion for this. 😅

Another idea: maybe just change the button to "login with git" or something?

---

Anyway, just curious other people's thoughts on it.

### PR DESCRIPTION
**Pull request by crapStone**
closes #154

Explanation why I chose that path is also in that issue.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -52,10 +52,12 @@ type config struct {
 	GitlabClientKey string `yaml:"gitlab.client-key" env:"OG_GITLAB_CLIENT_KEY"`
 	GitlabSecret    string `yaml:"gitlab.secret" env:"OG_GITLAB_SECRET"`
 	GitlabUrl       string `yaml:"gitlab.url" env:"OG_GITLAB_URL"`
+	GitlabName      string `yaml:"gitlab.service-name" env:"OG_GITLAB_SERVICE_NAME"`
 
 	GiteaClientKey string `yaml:"gitea.client-key" env:"OG_GITEA_CLIENT_KEY"`
 	GiteaSecret    string `yaml:"gitea.secret" env:"OG_GITEA_SECRET"`
 	GiteaUrl       string `yaml:"gitea.url" env:"OG_GITEA_URL"`
+	GiteaName      string `yaml:"gitea.service-name" env:"OG_GITEA_SERVICE_NAME"`
```
Comment (alyssadev) [inline internal/config/config.go:60]: For clarity perhaps it might be better to have these fields named GiteaServiceName to match the yaml field?

Comment (alyssadev) [reply internal/config/config.go:60]: I apologise, that was unclear. My exact suggestion is this:
```diff
diff --git a/docs/configuration/cheat-sheet.md b/docs/configuration/cheat-sheet.md
index 756cabc..46f7f6c 100644
--- a/docs/configuration/cheat-sheet.md
+++ b/docs/configuration/cheat-sheet.md
@@ -24,11 +24,11 @@
 | gitlab.client-key     | OG_GITLAB_CLIENT_KEY     | none                  | The client key for the GitLab OAuth application.
                                                               |
 | gitlab.secret         | OG_GITLAB_SECRET         | none                  | The secret for the GitLab OAuth application.
                                                               |
 | gitlab.url            | OG_GITLAB_URL            | `https://gitlab.com/` | The URL of the GitLab instance.

                                                               |
-| gitlab.service-name   | OG_GITLAB_SERVICE_NAME   | `GitLab`              | The name of the GitLab instance. It is displayed in the OAuth login button.
                                                                                                           |
+| gitlab.name           | OG_GITLAB_NAME           | `GitLab`              | The name of the GitLab instance. It is displayed in the OAuth login button.
                                                                                                           |
 | gitea.client-key      | OG_GITEA_CLIENT_KEY      | none                  | The client key for the Gitea OAuth application.
                                                               |
 | gitea.secret          | OG_GITEA_SECRET          | none                  | The secret for the Gitea OAuth application.
                                                               |
 | gitea.url             | OG_GITEA_URL             | `https://gitea.com/`  | The URL of the Gitea instance.

                                                               |
-| gitea.service-name    | OG_GITEA_SERVICE_NAME    | `Gitea`               | The name of the Gitea instance. It is displayed in the OAuth login button.
                                                                                                          |
+| gitea.name            | OG_GITEA_NAME            | `Gitea`               | The name of the Gitea instance. It is displayed in the OAuth login button.
                                                                                                          |
 | oidc.client-key       | OG_OIDC_CLIENT_KEY       | none                  | The client key for the OpenID application.
                                                               |
 | oidc.secret           | OG_OIDC_SECRET           | none                  | The secret for the OpenID application.    
                                                               |
 | oidc.discovery-url    | OG_OIDC_DISCOVERY_URL    | none                  | Discovery endpoint of the OpenID provider.
                                                               |
diff --git a/internal/config/config.go b/internal/config/config.go
index 3e77f71..5f3a65e 100644
--- a/internal/config/config.go
+++ b/internal/config/config.go
@@ -52,12 +52,12 @@ type config struct {
        GitlabClientKey string `yaml:"gitlab.client-key" env:"OG_GITLAB_CLIENT_KEY"`
        GitlabSecret    string `yaml:"gitlab.secret" env:"OG_GITLAB_SECRET"`
        GitlabUrl       string `yaml:"gitlab.url" env:"OG_GITLAB_URL"`
-       GitlabName      string `yaml:"gitlab.service-name" env:"OG_GITLAB_SERVICE_NAME"`
+       GitlabName      string `yaml:"gitlab.name" env:"OG_GITLAB_SERVICE_NAME"`

        GiteaClientKey string `yaml:"gitea.client-key" env:"OG_GITEA_CLIENT_KEY"`
        GiteaSecret    string `yaml:"gitea.secret" env:"OG_GITEA_SECRET"`
        GiteaUrl       string `yaml:"gitea.url" env:"OG_GITEA_URL"`
-       GiteaName      string `yaml:"gitea.service-name" env:"OG_GITEA_SERVICE_NAME"`
+       GiteaName      string `yaml:"gitea.name" env:"OG_GITEA_SERVICE_NAME"`

        OIDCClientKey    string `yaml:"oidc.client-key" env:"OG_OIDC_CLIENT_KEY"`
        OIDCSecret       string `yaml:"oidc.secret" env:"OG_OIDC_SECRET"`
```

**Comment by alyssadev** [comment]
I would also suggest this change I've made on my own instance 7e5d427600b53e659feae8014d8da064b724fa4e

**Comment by thomiceli** [comment]
Throwing a random idea : use the domain name of the instance to get the name instead of a new config

**Comment by alyssadev** [comment]
Perhaps this could be fetched automatically with the option to override. Any html page on a Gitea-compatible instance should have a meta property tag for `og:site_name`, i imagine we'd want to parse out the "Git with a cup of tea" thing that seems to be present unless explicitly removed, as in the Gitea instance I have my opengist instance connected to.

```html
<meta property="og:site_name" content="TWS Gitea: Git with a cup of tea">
```

**Comment by crapStone** [comment]
> I would also suggest this change I've made on my own instance [7e5d427](https://github.com/thomiceli/opengist/commit/7e5d427600b53e659feae8014d8da064b724fa4e)

I talked with a friend about that page, but it dind't cross my mind to add the new config fields there :laughing:

**Comment by crapStone** [comment]
> Throwing a random idea : use the domain name of the instance to get the name instead of a new config

I'm evaluating this service for Codeberg (see [here](https://codeberg.org/Codeberg/Community/issues/382)) and just using `codeberg.org` isn't enough, because we are not just another Forgejo/Gitea instance.

I could add a check after the parsing of the config and if no custom name is set the name will be set to the domain name.

File: docs/configuration/cheat-sheet.md [hunk 2]
```diff
@@ -24,11 +24,11 @@
...
 | gitea.url             | OG_GITEA_URL             | `https://gitea.com/`  | The URL of the Gitea instance.                                                                                                                                                                                                   |
-| gitea.service-name    | OG_GITEA_SERVICE_NAME    | `Gitea`               | The name of the Gitea instance. It is displayed in the OAuth login button.                                                                                                                                                                                                  |
+| gitea.name            | OG_GITEA_                | `Gitea`               | The name of the Gitea instance. It is displayed in the OAuth login button.                                                                                                                                                                                                  |
```
Comment (thomiceli) [inline docs/configuration/cheat-sheet.md:31]: -> `OG_GITEA_NAME`

**Comment by jolheiser** [comment]
fwiw I think this impl is fine. A domain check seems like it may be overcomplicating it when an override exists.

**Comment by thomiceli** [comment]
Thank you for the contributions :)

**Pull request merged**